rad/src/script.rs

```diff
@@ -20,27 +21,98 @@ use crate::{
 pub type RadonCall = (RadonOpCodes, Option<Vec<Value>>);
 pub type RadonScript = Vec<RadonCall>;
 
+/// A set of flags for telling the RADON executor what execution features to enable and what
+/// metadata to collect.
+#[derive(Clone, Copy)]
+pub struct RadonScriptExecutionSettings {
+    pub partial_results: bool,
+    pub timing: bool,
+    pub breakpoints: bool,
+}
+
+/// Default to enabling all execution features except `partial_results`.
+impl Default for RadonScriptExecutionSettings {
+    fn default() -> Self {
+        Self::all_but_partial_results()
+    }
+}
+
+impl RadonScriptExecutionSettings {
+    /// Enable all execution features except `partial_results`. This is the default for
+    /// `witnet_node`.
+    pub fn all_but_partial_results() -> Self {
+        Self {
+            partial_results: false,
+            ..Self::enable_all()
+        }
+    }
+
+    /// Disable all execution features.
+    pub fn disable_all() -> Self {
+        Self {
+            partial_results: false,
+            timing: false,
+            breakpoints: false,
+        }
+    }
+
+    /// Enable all execution features. This is recommended for `witnet_wallet`.
+    pub fn enable_all() -> Self {
+        Self {
+            partial_results: true,
+            timing: true,
+            breakpoints: true,
+        }
+    }
+}
+
 /// Run any RADON script on given input data, and return `RadonReport`.
 pub fn execute_radon_script(
     input: RadonTypes,
     script: &[RadonCall],
     context: &mut ReportContext,
+    settings: RadonScriptExecutionSettings,
 ) -> Result<RadonReport<RadonTypes>, RadError> {
     // Set the execution timestamp
-    context.start();
-    // Run the execution
+    if settings.timing {
+        context.start();
+    }
+
+    // Initialize a vector for storing the partial results
+    let mut partial_results = vec![Ok(input.clone())];
```

Comment:
You can avoid this clone when `setings.partial_results` is disabled, by setting partial_results to `None` here:

```suggestion
    let mut partial_results = if settings.partial_results { Some(vec![Ok(input.clone())]) } else { None };
```

And later using `if let Some(partial_results) = &mut partial_results` {}